sparta/sparta/statistics/StatisticInstance.hpp

```diff
@@ -1168,41 +560,41 @@ namespace sparta
          * compute its value. (nullptr if there is no StatisticDef to
          * reference)
          */
-        const StatisticDef* sdef_;
+        const StatisticDef* sdef_ = nullptr;
 
         /*!
          * \brief Counter reference from which this statistic instance will
          * compute its value (nullptr if there is no Counter to reference)
          */
-        const CounterBase* ctr_;
+        const CounterBase* ctr_ = nullptr;
 
         /*!
          * \brief Parameter reference from which this statistic instance will
          * compute its value (nullptr if there is no Parameter to reference)
          */
-        const ParameterBase* par_;
+        const ParameterBase* par_ = nullptr;
```

Comment:
It looks like these three members are never used at the same time. Either `sdef_` is used, or `ctr_` is used, or `par_` is used. If these don't share a common base class with virtual methods, it might make sense to add a templatized wrapper class that can handle this instead of having if-else-if-else-if everywhere.

Reply:
Yeah, I can see the original implementation could have used some better design.  I think this is also beyond the scope of this PR.  I'll append this to the existing issue I opened.